src/network/tc/htb.c

```diff
@@ -92,16 +95,80 @@ int config_parse_hierarchy_token_bucket_default_class(
         return 0;
 }
 
+int config_parse_hierarchy_token_bucket_u32(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(qdisc_free_or_set_invalidp) QDisc *qdisc = NULL;
+        HierarchyTokenBucket *htb;
+        Network *network = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = qdisc_new_static(QDISC_KIND_HTB, network, filename, section_line, &qdisc);
+        if (r == -ENOMEM)
+                return log_oom();
```

Comment:
We should handle the `log_oom()` at ` qdisc = new(QDisc, 1);` as it's not a library. Here we should handle error `r < 0`

Reply:
I will consider your point later in other PR. Thanks.